src/web/services/email-generation.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ResourceEndpoints } from '../types/api-endpoints';
+import { Email } from '../types/api-output';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles email generation.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class EmailGenerationService {
+
+  constructor(private httpRequestService: HttpRequestService) { }
+
+  /**
+   * Get email contents by calling API.
+   */
+  getEmail(courseId: string, studentemail: string, emailtype: string, fsname?: string): Observable<Email> {
```

Comment:
Although it works, it's not exactly the best way to generalize this method for all email types. Why not create a few different methods for different email types? i.e.
```ts
getCourseJoinEmail(courseId: string, studentEmail: string): Observable<Email> {
  ...
}

getFeedbackSessionReminderEmail(courseId: string, studentEmail: string): Observable<Email> {
  ...
}
```
This way you can also hide some implementation details (e.g. the `EmailType`) from the service callers.